libconnman-qt/networkmanager.cpp

```diff
@@ -746,13 +746,15 @@ void NetworkManager::updateServices(const ConnmanObjectList &changed, const QLis
     // Removed services
     for (const QDBusObjectPath &obj : removed) {
         const QString path(obj.path());
-        NetworkService *service = m_servicesCache.value(path);
+        NetworkService *service = m_servicesCache.take(path);
         if (service) {
             if (service == m_priv->m_connectedWifi) {
                 m_priv->m_connectedWifi = NULL;
             }
+            if (m_defaultRoute == service) {
+                m_defaultRoute = NULL;
```

Comment:
Just wonderin that should we just set this invalid like,
```
m_defaultRoute = m_invalidDefaultRoute
```
as that will be the end result.

Reply:
Done. Seems to work fine still. I just noticed that the previous if does a very similar thing, so I'll flip the operands so service is first in this one too.